### PR TITLE
Feat/revise import syntax

### LINKED ISSUE #143: Revise import syntax
Early experience with import syntax suggests some changes are in order.

| current | proposed |
|:-|:-|
| `//.str.lower("HELLO")` | `//str.lower("HELLO")` |
| `//github.com/"foo-bar".x.y` | `//{github.com/foo-bar}.x.y` |
| `//./foo.bar` | `//{./foo}.bar` |
| `///foo.bar` | `//{/foo}.bar` |

The primary motivations are:

1. Importing standard library packages is a very common operation and `//.x.y` is a bit tedious to type. `//x.y` is easier and cleaner.
2. Remote imports are currently awkward to express because paths might have elements such as `-` that correspond to arr.ai syntactic elements, thus requiring quoting to disambiguate. The `//{...}` syntax makes the package name unambiguous.
3. For similar reasons, it makes sense for local imports to follow suit.

### PR DESCRIPTION
**Pull request by tiagralee**
Fixes #143  .

Changes proposed in this pull request:
- update syntax gramma for pkg
- update related tests

Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

File: examples/grpc/app.arrai [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//./'grpc-proto'((
+//{./grpc-proto}((
```
Comment (tiagralee) [inline examples/grpc/app.arrai:1]: not sure if this change is correct for the new syntax

File: examples/grpc/grpc-proto.arrai [hunk 2]
```diff
@@ -1,9 +1,9 @@
 \app
-let grpc = //./grpc;
+let grpc = //{./grpc};
```
Comment (tiagralee) [inline examples/grpc/grpc-proto.arrai:2]: ditto

File: examples/grpc/grpc-proto.arrai [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 let wrap = "wrap" <: app.attrs.patterns;
-let proto = //./'proto-util'(wrap);
-let endpoints = app.endpoints where !//.str.has_prefix(.@item.name, "enum ");
-//.archive.tar.tar({
+let proto = //{./proto-util}(wrap);
```
Comment (tiagralee) [inline examples/grpc/grpc-proto.arrai:4]: ditto

File: examples/grpc/proto-util.arrai [hunk 4]
```diff
@@ -1,16 +1,16 @@
-let grpc = //./grpc;
+let grpc = //{./grpc};
```
Comment (tiagralee) [inline examples/grpc/proto-util.arrai:1]: ditto

File: syntax/parse.go [hunk 5]
```diff
@@ -90,9 +89,10 @@ sexpr  -> "${"
 
 ARROW  -> /{:>|=>|>>|orderby|order|where|sum|max|mean|median|min};
 IDENT  -> /{ \. | [$@A-Za-z_][0-9$@A-Za-z_]* };
+STRING -> /{ [$@A-Za-z_/][0-9$@A-Za-z_:.\/-]* };
 STR    -> /{ " (?: \\. | [^\\"] )* "
            | ' (?: \\. | [^\\'] )* '
-           | ‵ (?: ‵‵  | [^‵  ] )* ‵
+		   | ‵ (?: ‵‵  | [^‵  ] )* ‵
```
Comment (marcelocantos) [inline syntax/parse.go:95]:
```suggestion
           | ‵ (?: ‵‵  | [^‵  ] )* ‵
```

File: syntax/parse.go [hunk 6]
```diff
@@ -90,9 +89,10 @@ sexpr  -> "${"
 
 ARROW  -> /{:>|=>|>>|orderby|order|where|sum|max|mean|median|min};
 IDENT  -> /{ \. | [$@A-Za-z_][0-9$@A-Za-z_]* };
+STRING -> /{ [$@A-Za-z_/][0-9$@A-Za-z_:.\/-]* };
```
Comment (marcelocantos) [inline syntax/parse.go:92]:
```suggestion
PKGPATH -> /{ { (?: \\. | [^\\}] )* \} };
```

File: syntax/parse.go [hunk 7]
```diff
@@ -63,10 +63,9 @@ expr   -> C* amp="&"* @ C* arrow=(
         | C* "{:" C* embed=(grammar=@ ":" subgrammar=%%ast) ":}" C*
         | C* op="\\\\" @ C*
         | C* fn="\\" IDENT @ C*
-        | C* "//" pkg=( dot="."? ("/" name)+
-                      | "." std=IDENT?
-                      | http=/{https?://}? fqdn=name:"." ("/" path=name)*
-                      )
+		| C* "//" pkg=( "{" dot="."? STRING "}"
+				| std=IDENT?
+		)
```
Comment (marcelocantos) [inline syntax/parse.go:68]:
```suggestion
        | C* "//" pkg=( PKGPATH | std=IDENT? )
```
You'll need to update the logic to accommodate this.

File: syntax/parse.go (hunk 6 above)
Comment (tiagralee) [reply syntax/parse.go]: need to take the `.` out, there is an issue with this rule when trying to compile `{1,2,3,4} => .^2` looks like the . `^2` are separated and got issue with `^2` have no match rule.

**Pull request merged**